components/org.wso2.carbon.identity.oauth.dcr.endpoint/src/test/java/org/wso2/carbon/identity/oauth2/dcr/endpoint/impl/RegisterApiServiceImplTest.java

```diff
@@ -0,0 +1,181 @@
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.wso2.carbon.identity.oauth2.dcr.endpoint.impl;
+
+import org.mockito.Mock;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.testng.Assert;
+import org.testng.IObjectFactory;
+import org.testng.annotations.*;
+import org.wso2.carbon.identity.oauth.dcr.bean.Application;
+import org.wso2.carbon.identity.oauth.dcr.exception.DCRMClientException;
+import org.wso2.carbon.identity.oauth.dcr.exception.DCRMException;
+import org.wso2.carbon.identity.oauth.dcr.exception.DCRMServerException;
+import org.wso2.carbon.identity.oauth.dcr.service.DCRMService;
+import org.wso2.carbon.identity.oauth2.dcr.endpoint.dto.RegistrationRequestDTO;
+import org.wso2.carbon.identity.oauth2.dcr.endpoint.dto.UpdateRequestDTO;
+import org.wso2.carbon.identity.oauth2.dcr.endpoint.util.DCRMUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.powermock.api.mockito.PowerMockito.*;
+import static org.powermock.api.mockito.PowerMockito.when;
+
+@PrepareForTest({DCRMUtils.class, DCRMService.class})
+@PowerMockIgnore("javax.*")
+public class RegisterApiServiceImplTest {
+    private Application application = new Application();
+    private List<String> redirectUris = new ArrayList<>();
+    private RegisterApiServiceImpl registerApiService = new RegisterApiServiceImpl();
+    @Mock
+    private DCRMUtils dcrmUtils;
+
+    @Mock
+    private DCRMService dcrmService;
+
+    @BeforeMethod
+    public void setUp() throws Exception {
+        redirectUris.add("https://op.certification.openid.net:60845/authz_cb");
+        application.setClient_name("Application");
+        application.setClient_id("N2QqQluzQuL5X6CtM3KZwqzLQhUa");
+        application.setClient_secret("4AXWrN88aEfMvq2h_G0dN05KRsUa");
+        application.setRedirect_uris(redirectUris);
+        mockStatic(DCRMUtils.class);
+        mockStatic(DCRMService.class);
+        when(DCRMUtils.getOAuth2DCRMService()).thenReturn(dcrmService);
+    }
+
+    @DataProvider(name = "BuildDeleteApplication")
+    public Object[][] buildDeleteApplication() {
+        DCRMClientException dcrmClientException = new DCRMClientException("DCRMClientException");
+        dcrmClientException.setErrorCode("CONFLICT_");
+        return new Object[][] {
+                {"N2QqQluzQuL5X6CtM3KZwqzLQhUq", false, null},
+                {"N2QqQluzQuL5X6CtM3KZwqzLQyys", true, dcrmClientException},
+        };
+    }
+
+    @Test(dataProvider ="BuildDeleteApplication")
+    public void testDeleteApplication(String clientId, boolean isException, Object exception) throws DCRMException {
+        if (isException) {
```

Comment:
rather than branching inside the test and testing two different flows writing two test cases would be much cleaner IMO.
ie. One test case for success scenario and other for the scenario where an exception is thrown.

Reply:
Consider this change for other similar test cases.